src/toil/jobStores/aws/jobStore.py

```diff
@@ -307,7 +307,8 @@ def _extractKeyInfoFromUrl(url):
         """
         s3 = boto.connect_s3()
         bucket = s3.get_bucket(url.netloc)
-        key = bucket.new_key(url.path[1:])
+        key = bucket.get_key(url.path[1:])
+        key = key if key is not None else bucket.new_key(url.path[1:])
```

Comment:
Doing this creates an issue of having to mess around with the url again after ```_extractKeyInfoFromUrl()``` is called. 
e.g. 
```
dstBucket, newKey = self._extractKeyInfoFromUrl(url)
assert newKey is None
newKey = self.s3.new_key(url.path[1:])      
```
How about instead adding a parameter that explicitly specifies what should happen if the key does not already exist
```
_extractKeyInfoFromUrl(url, createKey)
```
This way the consideration for either case must be made with one boolean value rather than testing for None.